include/SDL3/SDL_dialog.h

```diff
@@ -0,0 +1,190 @@
+ * yet exist.
+ *
+ * \param callback The function to be invoked when the user selects a file and accepts, or the user cancels the dialog, or an error occurs. The first argument is a null-terminated list of C strings, representing the paths chosen by the user. The list will be empty if the user canceled the dialog, and it will be NULL if an error occured. If an error occured, it can be fetched with SDL_GetError(). The second argument is the userdata pointer passed to the function.
+ * \param userdata An optional pointer to pass extra data to the callback when it will be invoked.
+ * \param window The window to bind the dialog to. May be NULL. Not all platforms support this option.
+ * \param filters A null-terminated list of SDL_DialogFileFilter's. May be NULL. Not all platforms support this option, and platforms that do support it may allow the user to ignore the filters.
+ * \param default_location The default location to start the dialog at. May be NULL. Not all platforms support this option.
+ * \param allow_many If non-zero, the user will be allowed to select multiple entries. Not all platforms support this option.
+ *
+ * \since This function is available since SDL 3.0.0.
+ *
+ * \sa SDL_DialogFileFilter
+ * \sa SDL_DialogFileCallback
+ * \sa SDL_ShowSaveFileDialog
+ * \sa SDL_ShowOpenFolderDialog
+ */
+extern DECLSPEC void SDLCALL SDL_ShowOpenFileDialog(SDL_DialogFileCallback callback, void* userdata, SDL_Window* window, const SDL_DialogFileFilter *filters, const char* default_location, int allow_many);
+
+/**
+ * Displays a dialog that lets the user choose a new or existing file on their filesystem.
+ *
+ * This function should only be invoked from the main thread.
+ *
+ * This is an asynchronous function; it will return immediately, and the result
+ * will be passed to the callback.
+ *
+ * The callback will be invoked with a null-terminated list of files the user
+ * chose. The list will be empty if the user canceled the dialog, and it will
+ * be NULL if an error occured.
+ *
+ * Note that the callback may be called from a different thread than the one
+ * the function was invoked on.
+ *
+ * If an error occurs, the error can be obtained with SDL_GetError().
+ *
+ * The chosen file may or may not already exist.
+ *
+ * \param callback The function to be invoked when the user selects a file and accepts, or the user cancels the dialog, or an error occurs. The first argument is a null-terminated list of C strings, representing the paths chosen by the user. The list will be empty if the user canceled the dialog, and it will be NULL if an error occured. If an error occured, it can be fetched with SDL_GetError(). The second argument is the userdata pointer passed to the function.
+ * \param userdata An optional pointer to pass extra data to the callback when it will be invoked.
+ * \param window The window to bind the dialog to. May be NULL. Not all platforms support this option.
```

Comment:
What does binding a dialog to a window do?

Reply:
It makes the dialog modal. What's the correct word to use when talking about the window that is "bound" to a modal dialog?

Reply:
Saying that the dialog becomes modal for that window makes sense to me.